--- a/com/daml/ledger/api/v2/admin/command_inspection_service.proto
+++ b/com/daml/ledger/api/v2/admin/command_inspection_service.proto
@@ -10,6 +10,7 @@ import "com/daml/ledger/api/v2/completion.proto";
 import "com/daml/ledger/api/v2/value.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2/admin;adminv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2.Admin";
 option java_outer_classname = "CommandInspectionServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2.admin";
