--- a/com/daml/ledger/api/v2/admin/participant_pruning_service.proto
+++ b/com/daml/ledger/api/v2/admin/participant_pruning_service.proto
@@ -5,6 +5,7 @@ syntax = "proto3";
 
 package com.daml.ledger.api.v2.admin;
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2/admin;adminv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2.Admin";
 option java_outer_classname = "ParticipantPruningServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2.admin";
