--- a/com/daml/ledger/api/v2/interactive/interactive_submission_service.proto
+++ b/com/daml/ledger/api/v2/interactive/interactive_submission_service.proto
@@ -13,9 +13,11 @@ import "com/daml/ledger/api/v2/package_reference.proto";
 import "com/daml/ledger/api/v2/transaction.proto";
 import "com/daml/ledger/api/v2/transaction_filter.proto";
 import "com/daml/ledger/api/v2/value.proto";
+import "google/api/annotations.proto";
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2/interactive;interactivev2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2.Interactive";
 option java_outer_classname = "InteractiveSubmissionServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2.interactive";
@@ -30,15 +32,35 @@ option java_package = "com.daml.ledger.api.v2.interactive";
 // This gives callers the ability to sign the daml transaction with their own signing keys
 service InteractiveSubmissionService {
   // Requires `readAs` scope for the submitting party when LAPI User authorization is enabled
-  rpc PrepareSubmission(PrepareSubmissionRequest) returns (PrepareSubmissionResponse);
+  rpc PrepareSubmission(PrepareSubmissionRequest) returns (PrepareSubmissionResponse) {
+    option (google.api.http) = {
+      post: "/v2/interactive-submission/prepare"
+      body: "*"
+    };
+  }
   // Execute a prepared submission _asynchronously_ on the ledger.
   // Requires `actAs` or `executeAs` scope for the submitting party when LAPI User authorization is enabled
   // Requires a signature of the transaction from the submitting external party.
-  rpc ExecuteSubmission(ExecuteSubmissionRequest) returns (ExecuteSubmissionResponse);
+  rpc ExecuteSubmission(ExecuteSubmissionRequest) returns (ExecuteSubmissionResponse) {
+    option (google.api.http) = {
+      post: "/v2/interactive-submission/execute"
+      body: "*"
+    };
+  }
   // Similar to ExecuteSubmission but _synchronously_ wait for the completion of the transaction
-  rpc ExecuteSubmissionAndWait(ExecuteSubmissionAndWaitRequest) returns (ExecuteSubmissionAndWaitResponse);
+  rpc ExecuteSubmissionAndWait(ExecuteSubmissionAndWaitRequest) returns (ExecuteSubmissionAndWaitResponse) {
+    option (google.api.http) = {
+      post: "/v2/interactive-submission/executeAndWait"
+      body: "*"
+    };
+  }
   // Similar to ExecuteSubmissionAndWait but additionally returns the transaction
-  rpc ExecuteSubmissionAndWaitForTransaction(ExecuteSubmissionAndWaitForTransactionRequest) returns (ExecuteSubmissionAndWaitForTransactionResponse);
+  rpc ExecuteSubmissionAndWaitForTransaction(ExecuteSubmissionAndWaitForTransactionRequest) returns (ExecuteSubmissionAndWaitForTransactionResponse) {
+    option (google.api.http) = {
+      post: "/v2/interactive-submission/executeAndWaitForTransaction"
+      body: "*"
+    };
+  }
 
   // A preferred package is the highest-versioned package for a provided package-name
   // that is vetted by all the participants hosting the provided parties.
@@ -52,7 +74,11 @@ service InteractiveSubmissionService {
   // Can be accessed by any Ledger API client with a valid token when Ledger API authorization is enabled.
   //
   // Provided for backwards compatibility, it will be removed in the Canton version 3.4.0
-  rpc GetPreferredPackageVersion(GetPreferredPackageVersionRequest) returns (GetPreferredPackageVersionResponse);
+  rpc GetPreferredPackageVersion(GetPreferredPackageVersionRequest) returns (GetPreferredPackageVersionResponse) {
+    option (google.api.http) = {
+      get: "/v2/interactive-submission/preferred-package-version"
+    };
+  }
 
   // Compute the preferred packages for the vetting requirements in the request.
   // A preferred package is the highest-versioned package for a provided package-name
@@ -70,7 +96,12 @@ service InteractiveSubmissionService {
   // Can be accessed by any Ledger API client with a valid token when Ledger API authorization is enabled.
   //
   // Experimental API: this endpoint is not guaranteed to provide backwards compatibility in future releases
-  rpc GetPreferredPackages(GetPreferredPackagesRequest) returns (GetPreferredPackagesResponse);
+  rpc GetPreferredPackages(GetPreferredPackagesRequest) returns (GetPreferredPackagesResponse) {
+    option (google.api.http) = {
+      post: "/v2/interactive-submission/preferred-packages"
+      body: "*"
+    };
+  }
 }
 
 // Hints to improve cost estimation precision of a prepared transaction
@@ -314,16 +345,7 @@ message ExecuteSubmissionRequest {
   // If omitted, the participant will assume the configured maximum deduplication time.
   //
   // Optional
-  oneof deduplication_period {
-    // Specifies the length of the deduplication period.
-    // It is interpreted relative to the local clock at some point during the submission's processing.
-    // Must be non-negative. Must not exceed the maximum deduplication time.
-    google.protobuf.Duration deduplication_duration = 3;
-
-    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
-    // Must be a valid absolute offset (positive integer).
-    int64 deduplication_offset = 4;
-  }
+  DeduplicationPeriod deduplication_period = 3 [json_name = "deduplicationPeriod"];
 
   // A unique identifier to distinguish completions for different submissions with the same change ID.
   // Typically a random UUID. Applications are expected to use a different UUID for each retry of a submission
@@ -372,16 +394,7 @@ message ExecuteSubmissionAndWaitRequest {
   // If omitted, the participant will assume the configured maximum deduplication time.
   //
   // Optional
-  oneof deduplication_period {
-    // Specifies the length of the deduplication period.
-    // It is interpreted relative to the local clock at some point during the submission's processing.
-    // Must be non-negative. Must not exceed the maximum deduplication time.
-    google.protobuf.Duration deduplication_duration = 3;
-
-    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
-    // Must be a valid absolute offset (positive integer).
-    int64 deduplication_offset = 4;
-  }
+  DeduplicationPeriod deduplication_period = 3 [json_name = "deduplicationPeriod"];
 
   // A unique identifier to distinguish completions for different submissions with the same change ID.
   // Typically a random UUID. Applications are expected to use a different UUID for each retry of a submission
@@ -441,16 +454,7 @@ message ExecuteSubmissionAndWaitForTransactionRequest {
   // If omitted, the participant will assume the configured maximum deduplication time.
   //
   // Optional
-  oneof deduplication_period {
-    // Specifies the length of the deduplication period.
-    // It is interpreted relative to the local clock at some point during the submission's processing.
-    // Must be non-negative. Must not exceed the maximum deduplication time.
-    google.protobuf.Duration deduplication_duration = 3;
-
-    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
-    // Must be a valid absolute offset (positive integer).
-    int64 deduplication_offset = 4;
-  }
+  DeduplicationPeriod deduplication_period = 3 [json_name = "deduplicationPeriod"];
 
   // A unique identifier to distinguish completions for different submissions with the same change ID.
   // Typically a random UUID. Applications are expected to use a different UUID for each retry of a submission
