--- a/com/daml/ledger/api/v2/commands.proto
+++ b/com/daml/ledger/api/v2/commands.proto
@@ -9,6 +9,7 @@
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "CommandsOuterClass";
 option java_package = "com.daml.ledger.api.v2";
