--- a/com/daml/ledger/api/v2/commands.proto
+++ b/com/daml/ledger/api/v2/commands.proto
@@ -9,6 +9,7 @@ import "com/daml/ledger/api/v2/value.proto";
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "CommandsOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -17,10 +18,10 @@ option java_package = "com.daml.ledger.api.v2";
 message Command {
   // Required
   oneof command {
-    CreateCommand create = 1;
-    ExerciseCommand exercise = 2;
-    ExerciseByKeyCommand exercise_by_key = 4;
-    CreateAndExerciseCommand create_and_exercise = 3;
+    CreateCommand create = 1 [json_name = "CreateCommand"];
+    ExerciseCommand exercise = 2 [json_name = "ExerciseCommand"];
+    ExerciseByKeyCommand exercise_by_key = 4 [json_name = "ExerciseByKeyCommand"];
+    CreateAndExerciseCommand create_and_exercise = 3 [json_name = "CreateAndExerciseCommand"];
   }
 }
 
@@ -183,16 +184,7 @@ message Commands {
   // If omitted, the participant will assume the configured maximum deduplication time.
   //
   // Optional
-  oneof deduplication_period {
-    // Specifies the length of the deduplication period.
-    // It is interpreted relative to the local clock at some point during the submission's processing.
-    // Must be non-negative. Must not exceed the maximum deduplication time.
-    google.protobuf.Duration deduplication_duration = 5;
-
-    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
-    // Must be a valid absolute offset (positive integer) or participant begin (zero).
-    int64 deduplication_offset = 6;
-  }
+  DeduplicationPeriod deduplication_period = 5 [json_name = "deduplicationPeriod"];
 
   // Lower bound for the ledger time assigned to the resulting transaction.
   // Note: The ledger time of a transaction is assigned as part of command interpretation.
@@ -296,3 +288,20 @@ message PrefetchContractKey {
   // Optional
   optional uint32 limit = 3;
 }
+
+// Specifies the deduplication period for the change ID.
+// If omitted, the participant will assume the configured maximum deduplication time.
+//
+// Optional
+message DeduplicationPeriod {
+  oneof period {
+    // Specifies the length of the deduplication period.
+    // It is interpreted relative to the local clock at some point during the submission's processing.
+    // Must be non-negative. Must not exceed the maximum deduplication time.
+    google.protobuf.Duration deduplication_duration = 1 [json_name = "DeduplicationDuration"];
+
+    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
+    // Must be a valid absolute offset (positive integer) or participant begin (zero).
+    int64 deduplication_offset = 2 [json_name = "DeduplicationOffset"];
+  }
+}
