--- a/com/daml/ledger/api/v2/command_completion_service.proto
+++ b/com/daml/ledger/api/v2/command_completion_service.proto
@@ -7,7 +7,9 @@ package com.daml.ledger.api.v2;
 
 import "com/daml/ledger/api/v2/completion.proto";
 import "com/daml/ledger/api/v2/offset_checkpoint.proto";
+import "google/api/annotations.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "CommandCompletionServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -29,7 +31,12 @@ option java_package = "com.daml.ledger.api.v2";
 service CommandCompletionService {
   // Deprecated: please use ``GetCompletions`` instead.
   // Subscribe to command completion events.
-  rpc CompletionStream(CompletionStreamRequest) returns (stream CompletionStreamResponse);
+  rpc CompletionStream(CompletionStreamRequest) returns (stream CompletionStreamResponse) {
+    option (google.api.http) = {
+      post: "/v2/commands/completions"
+      body: "*"
+    };
+  }
 
   // Subscribe to command completion events.
   // This streaming endpoint provides more flexibility in filtering than the predecessor ``CompletionStream``.
@@ -83,8 +90,13 @@ message GetCompletionsRequest {
 
 message CompletionStreamResponse {
   // Required
-  oneof completion_response {
-    Completion completion = 1;
-    OffsetCheckpoint offset_checkpoint = 2;
+  CompletionResponse completion_response = 1 [json_name = "completionResponse"];
+}
+
+// Carries either a command completion or an offset checkpoint.
+message CompletionResponse {
+  oneof response {
+    Completion completion = 1 [json_name = "Completion"];
+    OffsetCheckpoint offset_checkpoint = 2 [json_name = "OffsetCheckpoint"];
   }
 }
