--- a/com/daml/ledger/api/v2/command_submission_service.proto
+++ b/com/daml/ledger/api/v2/command_submission_service.proto
@@ -7,7 +7,9 @@
 
 import "com/daml/ledger/api/v2/commands.proto";
 import "com/daml/ledger/api/v2/reassignment_commands.proto";
+import "google/api/annotations.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "CommandSubmissionServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -28,10 +30,20 @@
 // Clients SHOULD subscribe to the CompletionStream before starting to submit commands to prevent race conditions.
 service CommandSubmissionService {
   // Submit a single composite command.
-  rpc Submit(SubmitRequest) returns (SubmitResponse);
+  rpc Submit(SubmitRequest) returns (SubmitResponse) {
+    option (google.api.http) = {
+      post: "/v2/commands/async/submit"
+      body: "commands"
+    };
+  }
 
   // Submit a single reassignment.
-  rpc SubmitReassignment(SubmitReassignmentRequest) returns (SubmitReassignmentResponse);
+  rpc SubmitReassignment(SubmitReassignmentRequest) returns (SubmitReassignmentResponse) {
+    option (google.api.http) = {
+      post: "/v2/commands/async/submit-reassignment"
+      body: "*"
+    };
+  }
 }
 
 // The submitted commands will be processed atomically in a single transaction. Moreover, each ``Command`` in ``commands`` will be executed in the order specified by the request.
