--- a/com/daml/ledger/api/v2/event.proto
+++ b/com/daml/ledger/api/v2/event.proto
@@ -9,6 +9,7 @@ import "com/daml/ledger/api/v2/value.proto";
 import "google/protobuf/timestamp.proto";
 import "google/rpc/status.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "EventOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -28,9 +29,9 @@ message Event {
   oneof event {
     // The event as it appeared in the context of its original daml transaction on this participant node.
     // In particular, the offset, node_id pair of the daml transaction are preserved.
-    CreatedEvent created = 1;
-    ArchivedEvent archived = 2;
-    ExercisedEvent exercised = 3;
+    CreatedEvent created = 1 [json_name = "CreatedEvent"];
+    ArchivedEvent archived = 2 [json_name = "ArchivedEvent"];
+    ExercisedEvent exercised = 3 [json_name = "ExercisedEvent"];
   }
 }
 
@@ -80,7 +81,7 @@ message CreatedEvent {
   // The arguments that have been used to create the contract.
   //
   // Required
-  Record create_arguments = 6;
+  Record create_arguments = 6 [json_name = "createArgument"];
 
   // Opaque representation of contract create event payload intended for forwarding
   // to an API server as a contract disclosed as part of a command
