--- a/com/daml/ledger/api/v2/event_query_service.proto
+++ b/com/daml/ledger/api/v2/event_query_service.proto
@@ -7,7 +7,9 @@ package com.daml.ledger.api.v2;
 
 import "com/daml/ledger/api/v2/event.proto";
 import "com/daml/ledger/api/v2/transaction_filter.proto";
+import "google/api/annotations.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "EventQueryServiceOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -36,7 +38,12 @@ service EventQueryService {
   // No events will be returned for contracts that have been pruned because they
   // have already been archived before the latest pruning offset.
   // If the contract cannot be found for the request, or all the contract-events are filtered, a CONTRACT_EVENTS_NOT_FOUND error will be raised.
-  rpc GetEventsByContractId(GetEventsByContractIdRequest) returns (GetEventsByContractIdResponse);
+  rpc GetEventsByContractId(GetEventsByContractIdRequest) returns (GetEventsByContractIdResponse) {
+    option (google.api.http) = {
+      post: "/v2/events/events-by-contract-id"
+      body: "*"
+    };
+  }
 }
 
 message GetEventsByContractIdResponse {
