--- a/com/daml/ledger/api/v2/experimental_features.proto
+++ b/com/daml/ledger/api/v2/experimental_features.proto
@@ -5,6 +5,7 @@ syntax = "proto3";
 
 package com.daml.ledger.api.v2;
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "ExperimentalFeaturesOuterClass";
 option java_package = "com.daml.ledger.api.v2";
