--- a/com/daml/ledger/api/v2/offset_checkpoint.proto
+++ b/com/daml/ledger/api/v2/offset_checkpoint.proto
@@ -7,6 +7,7 @@ package com.daml.ledger.api.v2;
 
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "OffsetCheckpointOuterClass";
 option java_package = "com.daml.ledger.api.v2";
