--- a/com/daml/ledger/api/v2/package_reference.proto
+++ b/com/daml/ledger/api/v2/package_reference.proto
@@ -8,6 +8,7 @@
 import "google/protobuf/empty.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "PackageReferenceOuterClass";
 option java_package = "com.daml.ledger.api.v2";
