--- a/com/daml/ledger/api/v2/package_reference.proto
+++ b/com/daml/ledger/api/v2/package_reference.proto
@@ -5,9 +5,9 @@ syntax = "proto3";
 
 package com.daml.ledger.api.v2;
 
-import "google/protobuf/empty.proto";
 import "google/protobuf/timestamp.proto";
 
+option go_package = "github.com/digital-asset/canton/go/ledger/api/v2;ledgerapiv2";
 option csharp_namespace = "Com.Daml.Ledger.Api.V2";
 option java_outer_classname = "PackageReferenceOuterClass";
 option java_package = "com.daml.ledger.api.v2";
@@ -86,11 +86,20 @@ message VettedPackages {
 // The serial of last ``VettedPackages`` topology transaction on a given
 // participant and synchronizer.
 message PriorTopologySerial {
+  // The absence of any previous ``VettedPackages`` topology transaction.
+  message NoPrior {}
+
   // Optional
+  PriorTopologySerialSerial serial = 1 [json_name = "serial"];
+}
+
+// Selects whether a ``PriorTopologySerial`` asserts a previous transaction's
+// serial or the absence of any previous transaction.
+message PriorTopologySerialSerial {
   oneof serial {
     // Previous transaction's serial.
-    uint32 prior = 1;
+    uint32 prior = 1 [json_name = "Prior"];
     // No previous transaction exists.
-    google.protobuf.Empty no_prior = 2;
+    PriorTopologySerial.NoPrior no_prior = 2 [json_name = "NoPrior"];
   }
 }
